Drupal 11 compatibility for drupal/getjwtonlogin 2.0.3.

getjwtonlogin has no Drupal 11 release; 2.0.3 declares ^8 || ^9 || ^10. This is
the Project Update Bot's change from https://www.drupal.org/project/getjwtonlogin/issues/3430753
(merge request !4, status RTBC):
- core_version_requirement becomes ^10.1 || ^11;
- the unit test uses HttpKernelInterface::MAIN_REQUEST instead of the removed
  MASTER_REQUEST.
Plus one cleanup: remove the unused import of
Symfony\Component\HttpKernel\Event\FilterResponseEvent, a class removed in
Symfony 5. The subscriber itself already type-hints ResponseEvent.

Composer refuses the package on Drupal 11 because drupal.org's metadata says
drupal/core ^8 || ^9 || ^10. composer.json lists it in
extra.drupal-lenient.allowed-list (mglaman/composer-drupal-lenient) so it can be
installed; this patch then makes Drupal enable it. Drop both once upstream
releases a Drupal 11 version.

diff --git a/getjwtonlogin.info.yml b/getjwtonlogin.info.yml
index ecf1e44..f8017ef 100644
--- a/getjwtonlogin.info.yml
+++ b/getjwtonlogin.info.yml
@@ -2,6 +2,6 @@ name: getjwtonlogin
 type: module
 description: 'A module to get JWT on login'
 package: Development
-core_version_requirement: ^8 || ^9 || ^10
+core_version_requirement: ^10.1 || ^11
 dependencies:
   - jwt
diff --git a/src/EventSubscriber/JwtLoginSubscriber.php b/src/EventSubscriber/JwtLoginSubscriber.php
index 75f453b..2fc0988 100644
--- a/src/EventSubscriber/JwtLoginSubscriber.php
+++ b/src/EventSubscriber/JwtLoginSubscriber.php
@@ -3,7 +3,6 @@
 namespace Drupal\getjwtonlogin\EventSubscriber;
 
 use Symfony\Component\EventDispatcher\EventSubscriberInterface;
-use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
 use Symfony\Component\HttpKernel\KernelEvents;
 use Drupal\Component\Serialization\Json;
 
diff --git a/tests/src/Unit/EventSubscriber/JwtLoginSubscriberTest.php b/tests/src/Unit/EventSubscriber/JwtLoginSubscriberTest.php
index bfde1ad..e5718ee 100644
--- a/tests/src/Unit/EventSubscriber/JwtLoginSubscriberTest.php
+++ b/tests/src/Unit/EventSubscriber/JwtLoginSubscriberTest.php
@@ -58,7 +58,7 @@ class JwtLoginSubscriberTest extends UnitTestCase {
     $request = new Request();
     $request->attributes->set('_route', 'user.logout.http');
     $response = new Response('{}', 200);
-    $event = new \Symfony\Component\HttpKernel\Event\ResponseEvent($this->httpKernel, $request, HttpKernelInterface::MASTER_REQUEST, $response);
+    $event = new \Symfony\Component\HttpKernel\Event\ResponseEvent($this->httpKernel, $request, HttpKernelInterface::MAIN_REQUEST, $response);
 
     // Call the event handler.
     $subscriber->onHttpLoginResponse($event);
@@ -79,7 +79,7 @@ class JwtLoginSubscriberTest extends UnitTestCase {
     $request = new Request();
     $request->attributes->set('_route', 'user.login.http');
     $response = new Response('{}', 400);
-    $event = new \Symfony\Component\HttpKernel\Event\ResponseEvent($this->httpKernel, $request, HttpKernelInterface::MASTER_REQUEST, $response);
+    $event = new \Symfony\Component\HttpKernel\Event\ResponseEvent($this->httpKernel, $request, HttpKernelInterface::MAIN_REQUEST, $response);
 
     // Call the event handler.
     $subscriber->onHttpLoginResponse($event);
@@ -100,7 +100,7 @@ class JwtLoginSubscriberTest extends UnitTestCase {
     $request = new Request();
     $request->attributes->set('_route', 'user.login.http');
     $response = new Response(JSON::encode(['current_user' => ['uid' => '1']]), 200);
-    $event = new \Symfony\Component\HttpKernel\Event\ResponseEvent($this->httpKernel, $request, HttpKernelInterface::MASTER_REQUEST, $response);
+    $event = new \Symfony\Component\HttpKernel\Event\ResponseEvent($this->httpKernel, $request, HttpKernelInterface::MAIN_REQUEST, $response);
 
     // Call the event handler.
     $subscriber->onHttpLoginResponse($event);
